src/adapter/src/coord/timestamp_oracle/postgres_oracle.rs

```diff
@@ -161,7 +161,7 @@ pub struct DynamicConfig {
     pg_connection_pool_max_size: AtomicUsize,
 
     /// The maximum time to wait when attempting to obtain a connection from the pool.
-    pg_connection_pool_max_wait: RwLock<Option<Duration>>,
```

Comment:
Hmm looks like this is intentionally an Option at the request of @pH14 https://github.com/MaterializeInc/materialize/pull/21955#discussion_r1337696745